### ISSUE TITLE
Simple webpage

### ISSUE THREAD

**Issue by betatim** [betatim/build-a-binder#11]
Before we can advertise things we need a simple webpage that lists the details for a workshop and at minimum links to a google form for registration.

Can someone recommend a nice and simple template for a webpage? (The standard ones that come with GitHub pages are a bit weird I think.)

**Comment by gedankenstuecke** [comment]
I just yesterday looked into the nice template that @aath0 started using for her GH page. Have a look here: https://github.com/aath0/aath0.github.io

I think this could also work nicely for a workshop page. Otherwise we could think about using one of the WOW workshop themes?

**Comment by pherterich** [comment]
What could probably also work is something similar to the software carpentry stuff as the workshop might be similar? <https://github.com/swcarpentry/workshop-template> Not really skilled enough to have a proper opinion though.

**Comment by betatim** [comment]
The https://github.com/aath0/aath0.github.io theme looks nice.

Good point about the workshop content itself. I hadn't thought as far as that yet. Was more thinking of a "landing page" for registration and info.

I like the carpentry workshop template, the last time I suggested it though lots of people voted for GitBook instead. Setting up the carpentry template is quick but you end up customising, GitBook takes a moment as well :-/ Maybe someone who has opinions/feelings on this can take the lead :)

The minimal effort part of me is thinking: can we get away with example repos that demonstrate how to do X, markdown files in this repo, and slides on google docs (or similar)?

**Comment by trallard** [comment]
My thoughts:

- Simple landing page with workshop basic info: venues, dates, code of conduct, links to registration, contact details, etc. 
- As for the materials themselves, we could follow any of the approaches:
  - Gitbook: could have everything in simple markdown and get them rendered in gitbook
  - Jupyter notebook: have all the contents as Jupyter notebook and could set up a basic web version from these
  - Sphinx: content could be in the form of a Jupyter notebook or .md files

I have used all three approaches for previous courses and I could definitely help with this if you needed me to. So basically it reduces to which format do you want to write the materials in and then translating them into a user-friendly nice looking output

**Comment by betatim** [comment]
I really don't know what format is best. I'd pick one based on what someone likes and then see how it goes. Hadn't thought of sphinx but now that you bring it up I realise that most (if not all) the existing binder docs and material are sphinx based. So maybe that is what tips it? We could also use that as landing page (reducing the number of techs by factor two in one fell swoop!)

Do you want to give that a go?

**Comment by trallard** [comment]
Using that as landing page and workshop materials source will definitely reduce the work.

I will have a crack at putting together a draft site (probably on Wednesday or Thursday). 

And will ship it in the form of a PR first so that you can review it first and can sort out additional privileges later on if needed.

**Comment by betatim** [comment]
I look forward to bookmarking https://build-a-binder.readthedocs.io :)

**Comment by willingc** [comment]
@betatim You might want to check out Hugo too. Some great themes many with excellent support, fast builds and deploys especially when using Netlify,

**Comment by RaoOfPhysics** [comment]
Late to the party and don't want to derail plans for a sphinx-based website, but for a landing page, you might want to check out https://github.com/gdg-x/zeppelin which @neuroamanda _et al._ used for their event landing page recently: https://japanscicom.github.io/ It needs… a little love to customise.

I would suggest using Jekyll over Hugo and others as it plugs straight in to GitHub allowing anyone to make a small change via PR without rebuilding the website themselves. But I have no strong feelings about this!

I support the idea of having a separate landing page and a separate page for hosting workshop content.

**Comment by willingc** [comment]
Cool sites @RaoOfPhysics.

Both Jekyll and Hugo as well as others are good options. I'm not sure that I understand the statement about using GitHub for small changes via PR with Jekyll vs Hugo. On other projects that I maintain, it's straightforward to edit content using the Github UI with Hugo. Combined with Netlify, we also are able to set up a preview of the PR when submitted (I believe with Netlify you can do that with Jekyll too).

Great to see these workshops moving forward. My general view on static site generators is use what folks feel comfortable with, get it up and running, and then focus most of your effort on the content :smile: :sunny:

**Comment by trallard** [comment]
I have no idea how I completely forgot about Jekyll. 

I suppose we could have a Jekyll page for the workshop logistics and the sphinx one for the workshop content.

Either way, both can be set up so that the site is built and tested when pushed/merged into GitHub. 

I am happy either way and neither should take me much time to set up

**Comment by RaoOfPhysics** [comment]
I could've been clearer, @willingc. :) I hadn't considered the use of GH Pages with Netlify, so I was pointing out that GH would automatically build a new Jekyll page when a change is pushed, without needing to build the website locally or using a CI tool. But I'm no expert, so my take might not be the best. :)

**Comment by gedankenstuecke** [comment]
I just wanted to 👍 @RaoOfPhysics's suggestion for having some non-sphinx page for the workshop logistics/advertisement page. Sphinx is great for docs but not the most inviting thing for getting people to register I fear. 😄

**Comment by trallard** [comment]
Ok so I have a draft for the site here :  [http://bitsandchips.me/build-a-binder-website/](http://bitsandchips.me/build-a-binder-website/) 

Some things to note:
- I have no idea what to write down in the 'about the workshop' section so I just left it with a mock text
- About the locations section: I would like to add an individual page per site including the exact location (with a map) directions to get there, and a local point of contact
- Also, it would be good to add the code of conduct to the site as well (see #7 )

If there are any other sections/information that you might want to add let me know. 

I imagine you'd like to add some sort of wording indicating that you got support from Mozilla for this, so I can add this and the logo in the about section.

Regarding the actual workshop content, we can definitely have it as a Sphinx thing 🦄

**Comment by betatim** [comment]
"You will learn how others are using Binder and related tools. At the end of the workshop you will be able to take some of your content (in a R or Jupyter notebook) and prepare it so that it can be used by others by clicking one link."

**Comment by betatim** [comment]
Drafting the signup form in #14

**Comment by heluc** [comment]
@betatim  Yes, room at EPFL will be BC 410. You can use this link: https://plan.epfl.ch/?room=BC410

**Comment by pherterich** [comment]
@betatim at Birmingham, we'll be in the Murray Learning Centre (R28 on this map https://www.birmingham.ac.uk/Documents/university/edgbaston-campus-map.pdf) in room UG05

**Comment by betatim** [comment]
@trallard do you want to create PR for what you already have? Then others can help fill out location, etc.

I think we should start advertising and collecting sign ups for the Birmingham event soon.

**Comment by betatim** [comment]
> I imagine you'd like to add some sort of wording indicating that you got support from Mozilla for this, so I can add this and the logo in the about section.

https://github.com/betatim/build-a-binder#support is the sentence that the funders indicated they want to see. There were no details about use/not use of logos etc As a version 1 I would include just that sentence.

**Comment by trallard** [comment]
I could create a PR with the website as it is now but I have not created the site pages yet. I am teaching all day (9-5:30) today until Friday so I am afraid I would only be able to work on the website on Saturday afternoonish though

**Comment by trallard** [comment]
I am working on the website... @betatim any idea on the location/date for the event in Germany? Will this be in CODE as discussed in #4 ?

**Comment by betatim** [comment]
It isn't locked in yet so I would leave it open. IMO this isn't something we need to wait for before starting to use the website.

**Comment by trallard** [comment]
Excellent! Do you have a link to the registration form for B'ham. So that I can add it as I presume the registration will be open now/soon?

Also when is the registration opening for Switzerland?
